packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1967,6 +2023,71 @@ export function writeEndSegment(
   }
 }
 
+export function writeCompletedSegmentInstruction(
+  destination: Destination,
+  responseState: ResponseState,
+  contentSegmentID: number,
+): boolean {
+  const containerID = stringToChunk(
+    responseState.segmentPrefix + contentSegmentID.toString(16),
+  );
+  const placeholderID = stringToChunk(
+    responseState.placeholderPrefix + contentSegmentID.toString(16),
+  );
+  return responseState.instructionWriter.writeCompletedSegmentInstruction({
+    write: chunk => writeChunkAndReturn(destination, chunk),
```

Comment:
Not sure what the perf implications of this are in JS. If there is nontrivial overhead, happy to change this to just pass destination as an argument